code/datums/components/full_auto_fire.dm

```diff
@@ -0,0 +1,357 @@
+#define AUTOFIRE_STAT_SLEEPING 0 //Component is in the gun, but the gun is in a different firemode. Sleep until a compatible firemode is activated.
+// ^^ sleep_up() ^^
+// VV wake_up() VV
+#define AUTOFIRE_STAT_IDLE 1 //Compatible firemode is in the gun. Wait until it's held in the user hands.
+// ^^ autofire_off() ^^
+// VV autofire_on() VV
+#define AUTOFIRE_STAT_ALERT	2 //Gun is active and in the user hands. Wait until user does a valid click.
+// ^^ stop_autofiring() ^^
+// VV start_autofiring() VV
+#define AUTOFIRE_STAT_FIRING 3 //Dakka-dakka-dakka.

```

Comment:
If these aren't going to be in a defines folder you need to undef them at the end of the file